<launch>

  <node pkg="tf" type="static_transform_publisher" name="base_to_camera" args="0.185 0.00 0.1 0 0 0   base_footprint camera_link 100" />

  <include file="$(find stella_3d_navigation)/launch/include/teb_local_planner_pure3d.launch">
  </include>

  <node pkg="tf" type="static_transform_publisher" name="base_to_depth2laser1" args="0.34 0.00 0.5 0 0 0  base_footprint laser1 100" />

  <node name="depth2laser1" pkg="depthimage_to_laserscan" type="depthimage_to_laserscan"  args= "standalone depthimage_to_laserscan/DepthImageToLaserScanNodelet">
    <remap from="image" to="/camera/depth/image"/>
    <remap from="camera_info" to="/camera/depth/camera_info"/>

    <param name="range_min" type="double" value="0.45"/>
    <param name="range_max" type="double" value="4"/>
    <param name="scan_height" type="int" value="10"/>
    <param name="output_frame_id" type="str" value="laser1"/>

    <remap from="scan" to="/scan"/>
  </node>


  <node pkg="tf" type="static_transform_publisher" name="base_to_depth2laser2" args="0.34 0.00 0.1 0 0 0  base_footprint laser2 100" />

  <node name="depth2laser2" pkg="depthimage_to_laserscan" type="depthimage_to_laserscan"  args= "standalone depthimage_to_laserscan/DepthImageToLaserScanNodelet">
    <remap from="image" to="/camera/depth/image"/>
    <remap from="camera_info" to="/camera/depth/camera_info"/>

    <param name="range_min" type="double" value="0.45"/>
    <param name="range_max" type="double" value="4"/>
    <param name="scan_height" type="int" value="250"/>
    <param name="output_frame_id" type="str" value="laser2"/>

    <remap from="scan" to="/scan2"/>
  </node>

  <!-- rtabmap -->
  <group ns="rtabmap">
      <arg name="database_path" default="$(find stella_3d_slam)/map_database/rtabmap.db"/>

      <arg name="localization" default="true"/>
      <arg     if="$(arg localization)" name="rtabmap_args" default=""/>
      <arg unless="$(arg localization)" name="rtabmap_args" default="-d"/>

    <node pkg="nodelet" type="nodelet" name="rgbd_sync" args="standalone rtabmap_ros/rgbd_sync" output="screen">
      <remap from="rgb/image"       to="/camera/rgb/image_raw"/>
      <remap from="depth/image"     to="/camera/depth/image"/>
      <remap from="rgb/camera_info" to="/camera/rgb/camera_info"/>
      <remap from="rgbd_image"      to="rgbd_image"/> <!-- output -->
      
      <!-- Should be true for not synchronized camera topics 
          (e.g., false for kinectv2, zed, realsense, true for xtion, kinect360)-->
      <param name="approx_sync"       value="true"/> 
    </node>

    <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="$(arg rtabmap_args)">
      <param name="frame_id" type="string" value="base_footprint"/>
      <param name="database_path"       type="string" value="$(arg database_path)"/>
        
      <param name="subscribe_depth" type="bool" value="false"/>
      <param name="subscribe_rgb" type="bool" value="false"/>
      <param name="subscribe_rgbd" type="bool" value="true"/>

      <remap from="odom" to="/odom"/>
      <remap from="rgbd_image" to="rgbd_image"/>
      <param name="queue_size" type="int" value="10"/>

      <!-- RTAB-Map's parameters -->
      <param name="RGBD/AngularUpdate"        type="string" value="0.01"/>
      <param name="RGBD/LinearUpdate"         type="string" value="0.01"/>
      <param name="RGBD/OptimizeFromGraphEnd" type="string" value="false"/>

      <!-- localization mode -->
      <param     if="$(arg localization)" name="Mem/IncrementalMemory" type="string" value="false"/>
      <param unless="$(arg localization)" name="Mem/IncrementalMemory" type="string" value="true"/>

    </node>

  </group>

  <node pkg="tf" type="static_transform_publisher" name="odom_to_map" args="0 0 0 0 0 0  map odom 100" />

  <node name='send_mark' pkg="stella_3d_navigation" type="send_mark.py">
  </node>

</launch>
